files/galaxy/tpv/tool_defaults.yml

```diff
@@ -60,6 +60,23 @@ tools:
           retval
         fail: |
           Invalid 'Remote resources id' selected in the config menu under 'User -> Preferences -> Manage Information -> Use distributed compute resources'. Please reselect either 'default' or an appropriate remote resource then click 'Save' and rerun your job.
+      - id: climate_strike
+        if: True
+        execute: |
+          from datetime import datetime
+          strike_start = datetime(2024,9,20,7,0)
+          strike_end = datetime(2024,9,20,20,0)
+          training_roles = (
+              [role.name for role in user.all_roles()
+              if not role.deleted and role.name in
+              ("training-bma231-ht24", "training-oubb")]
```

Comment:
can we do that more generic? startswith?

Reply:
We need to match the exact training role names. If a user runs a job and this user is part of the training role (and the specific training is still going on), then their jobs will continue to run. To do so, we need an exact match.

Reply:
From what I understand the edge case would be a user that has still or already a `training-*` role, but there is no training VM for them, right?

Reply:
If a user has a training role (training that is not active or old) and we do not follow the exact match rule, the job will be scheduled to the `compute` group (all users get the `compute` group through [this rule](https://github.com/usegalaxy-eu/infrastructure-playbook/blob/master/files/galaxy/tpv/tool_defaults.yml#L24-L35)) and will not be deferred. To avoid that, we do an exact match with the ongoing training and only allow the user job to run if they match; otherwise, they will be deferred.